connector/connect/src/main/scala/org/apache/spark/sql/connect/dsl/package.scala

```diff
@@ -238,13 +238,17 @@ package object dsl {
       def join(
           otherPlan: proto.Relation,
           joinType: JoinType = JoinType.JOIN_TYPE_INNER,
+          usingColumns: Seq[String] = Seq(),
```

Comment:
let's add an overload of `def join` which only accepts `usingColumns`, no `condition`

Reply:
Looks like only one overloaded version can have default arguments:

[ERROR] /Users/rui.wang/Documents/open-source-spark/spark/connector/connect/src/main/scala/org/apache/spark/sql/connect/dsl/package.scala:176: in class DslLogicalPlan, multiple overloaded alternatives of method join define default arguments.


I ended up with adding a few more overloaded version of `def join`.